From eaf12c514fc97d287c8cf678b71c4d6fbd8f782b Mon Sep 17 00:00:00 2001
From: Wadim Egorov <w.egorov@phytec.de>
Date: Thu, 23 Dec 2021 17:57:08 +0100
Subject: [PATCH] HACK: board: phycore_am64xx: Add Set CLKOUT0 to 25MHz

phyBOARD-Electra DP83867 PHYs require an 25 MHz clock input.
Configure RMII Clock Output to 25 MHz.
This change is board specific, does not belong to the SoM
initialization and therefore marked as a hack.

Signed-off-by: Wadim Egorov <w.egorov@phytec.de>
---
 board/phytec/phycore_am64xx/phycore-am64xx.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/board/phytec/phycore_am64xx/phycore-am64xx.c b/board/phytec/phycore_am64xx/phycore-am64xx.c
index 76d1621e40..678708923a 100644
--- a/board/phytec/phycore_am64xx/phycore-am64xx.c
+++ b/board/phytec/phycore_am64xx/phycore-am64xx.c
@@ -57,5 +57,11 @@ void spl_board_init(void)
 	val = readl(CTRLMMR_USB0_PHY_CTRL);
 	val &= ~(CORE_VOLTAGE);
 	writel(val, CTRLMMR_USB0_PHY_CTRL);
+
+	/* HACK: set clkout0 to 25MHz and enable */
+	/* 25MHz clock is needed for PRU-ICSS-G0 ETH PHYs */
+	writel(0x68ef3490, 0x43009008); /* kick0 */
+	writel(0xd172bc5a, 0x4300900c); /* kick1 */
+	writel(0x11, 0x43008010);       /* CLK_EN = 1, CLK_SEL = 1 */
 }
 #endif
-- 
2.25.1

